zcash_primitives/src/transaction/components/transparent/pczt.rs

```diff
@@ -0,0 +1,217 @@
+//! PCZT support for transparent Zcash.
+
+use std::collections::BTreeMap;
+
+use bip32::ChildNumber;
+use getset::Getters;
+use zcash_protocol::value::Zatoshis;
+
+use crate::{
+    legacy::Script,
+    transaction::{sighash::SighashType, TxId},
+};
+
+mod parse;
+pub use parse::ParseError;
+
+mod updater;
+pub use updater::{InputUpdater, OutputUpdater, Updater, UpdaterError};
+
+mod signer;
+pub use signer::SignerError;
+
+mod spend_finalizer;
+pub use spend_finalizer::SpendFinalizerError;
```

Comment:
Should these be behind "transparent-inputs" to avoid an unconditional dependency on secp256k1 and ripemd? They don't currently build without it.
```suggestion
#[cfg(feature = "transparent-inputs")]
mod updater;
#[cfg(feature = "transparent-inputs")]
pub use updater::{InputUpdater, OutputUpdater, Updater, UpdaterError};

#[cfg(feature = "transparent-inputs")]
mod signer;
#[cfg(feature = "transparent-inputs")]
pub use signer::SignerError;

#[cfg(feature = "transparent-inputs")]
mod spend_finalizer;
#[cfg(feature = "transparent-inputs")]
pub use spend_finalizer::SpendFinalizerError;
```

Reply:
I'm going with option 1. The `transparent-inputs` feature flag is for removing the `secp256k1` dependency, and transparent signing has a hard dependency on it (finer-grained would mean there's nothing actually left in the `signer` module).